am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -122,4 +126,36 @@ public FilterResourceResponse filterResource(String userId, String resourceId, J
 
         return null;
     }
+
+
+    /**
+     * <p>
+     * Method requires name of service, the type of resource, name of resource and a role name to
+     * populate a  map of JsonPointers and set of permissions when record with user role and type exists in database.
+     * </p>
+     *
+     * @param serviceName  name of service
+     * @param resourceType type of resource
+     * @param resourceName name of a resource
+     * @param roleNames    A set of role names. Currently only one role name is supported but
+     *                     in future implementations we shall support having multiple role names.
+     * @return permissionsByTypeAndRole a map of attributes and their corresponding permissions or null.
+     */
+    public Map<JsonPointer, Set<Permission>> getRolePermissions(
+        @NonNull String serviceName, @NonNull String resourceType,
+        @NonNull String resourceName, @NonNull Set<String> roleNames) {
+        if (roleNames.size() > MAX_ROLE_NAMES_RETURNED) {
+            throw new IllegalArgumentException();
```

Comment:
@rebeccabrennan  I asked you to communicate what happened better. Throwing exception without message is not helpful.

Reply:
👍